cmd/apiserver/start.go

```diff
@@ -30,8 +30,8 @@ import (
 
 	"github.com/jetstack-experimental/navigator/pkg/apis/navigator/v1alpha1"
 	"github.com/jetstack-experimental/navigator/pkg/apiserver"
-	clientset "github.com/jetstack-experimental/navigator/pkg/client/clientset_generated/internalclientset"
-	informers "github.com/jetstack-experimental/navigator/pkg/client/informers_generated/internalversion"
+	clientset "github.com/jetstack-experimental/navigator/pkg/client/clientset/internalversion"
+	informers "github.com/jetstack-experimental/navigator/pkg/client/informers/internalversion"
```

Comment:
❓ So how do we decide whether to use the `internalversion` or the `versioned` APIs? I guess I can see why the API server its self should use the internal versions, but why shouldn't the controller also use the internal APIs?

Reply:
Internal types should *only* be used within the API machinery for the apiserver, or if for *some reason* you need to convert between two external versions. That said, our controller should be built against one and only one version of the API, so there should basically never be a requirement to convert locally.

The controller shouldn't use the internal types for a number of reasons:

* These types can change, breaking compatibility.
* If we were to use the internal types, we'd then need to convert back into a versioned type in order to submit it back to the API server. We'd also need to convert *to* the internal version clientside too.
* I'd question what we have to gain from trying to do this
* It's also against all best practices for this sort of stuff, and would be a departure from how it is done within the k8s codebase. I can't recall an issue or justification for this decision, but I'm certain there is one with good reason.

There are a number of *other* potential complexities in doing this, and @dippynark and I did discuss it recently. I've managed to forget most of our thoughts on it though now! I will update here if they come back to me 😄